app/controllers/UkimsNumberController.scala

```diff
@@ -104,4 +107,45 @@ class UkimsNumberController @Inject() (
             } yield Redirect(navigator.nextPage(UkimsNumberUpdatePage, mode, updatedAnswers))
         )
   }
+
+  def onPageLoadCreateNew(mode: Mode): Action[AnyContent] =
+    (identify andThen getData andThen requireData) { implicit request =>
+      val preparedForm = request.userAnswers.get(UkimsNumberPage) match {
+        case None        => form
+        case Some(value) => form.fill(value)
+      }
+
+      Ok(newView(preparedForm, routes.UkimsNumberController.onSubmitCreateNew(mode), isCreateJourney = true))
+    }
+
+  def onSubmitCreateNew(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            Future
+              .successful(BadRequest(newView(formWithErrors, routes.UkimsNumberController.onSubmitCreateNew(mode)))),
+          value =>
+            for {
+              profile        <- traderProfileConnector.getTraderProfile(request.eori)
+              updatedAnswers <- Future.fromTry(request.userAnswers.set(UkimsNumberPage, value))
+              _              <- sessionRepository.set(updatedAnswers)
+            } yield
+              if (value == profile.ukimsNumber) {
+                val formWithApiErrors =
+                  createFormWithErrors(form, value, "ukimsNumberChangeController.duplicateUkimsNumber")
+                BadRequest(view(formWithApiErrors, routes.UkimsNumberController.onSubmitCreateNew(mode)))
+              } else {
+                Redirect(navigator.nextPage(UkimsNumberPage, mode, updatedAnswers))
+              }
```

Comment:
Is this redirect correct as , in the Navigator this take users to the HasNirmsController, which will already have been asked by now. Should it not redirect to the homepage?

Reply:
There is another [ticket](https://jira.tools.tax.service.gov.uk/browse/TGP-2700) that's supposed to handle the navigation for this view with its journey context